code/game/machinery/fax_machine.dm

```diff
@@ -1,9 +1,319 @@
+
+	if(!network || !uplink)
+		return FALSE
+
+	var/department_networks = LAZYACCESS(local_networks, network)
+	if(!department_networks)
+		local_networks[network] = list()
+		department_networks = local_networks[network]
+	var/datum/uplink/remote = LAZYACCESS(uplinks, uplink)
+	var/remote_networks = remote.networks
+
+	return list("[network]" = department_networks, "[uplink]" = remote_networks)
+
+/**
+ * Registers a department to a network.
+ *
+ * Arguments:
+ * network: required, the network to add the department to
+ * department: required, the department to add
+ */
+/datum/communications_network/proc/register_department(network, department)
+	LAZYADDASSOCLIST(local_networks, network, department)
+
+
+/**
+ * Unregisters a department from the a network.
+ *
+ * Arguments:
+ * network: required, the network to remove the department from
+ * department: required, the department to remove
+ */
+/datum/communications_network/proc/unregister_department(network, department)
+	LAZYREMOVEASSOC(local_networks, network, department)
+
+/datum/communications_network/proc/send_fax(sent, sent_name, mob/sender, target_department, network, obj/structure/machinery/faxmachine/origin, datum/uplink/authoriser)
+	SEND_SIGNAL(src, COMSIG_FAX_SENT, target_department, network, sent, sent_name, sender, origin, authoriser)
+
+/**
+ * Announces a fax to ghosts and admins.
+ *
+ * Arguments:
+ * msg_admin: required, the message to send to admins
+ * msg_ghost: optional, the message to send to ghosts
+ */
+/datum/communications_network/proc/announce_fax(msg_admin, msg_ghost)
+	log_admin(msg_admin) //Always irked me the replies do show up but the faxes themselves don't
+	for(var/client/C in GLOB.admins)
+		if((R_ADMIN|R_MOD) & C.admin_holder.rights)
+			if(msg_admin)
+				to_chat(C, msg_admin)
+			else
+				to_chat(C, msg_ghost)
+			SEND_SOUND(C, 'sound/effects/sos-morse-code.ogg')
+	if(msg_ghost)
+		for(var/i in GLOB.observer_list)
+			var/mob/dead/observer/g = i
+			if(!g.client)
+				continue
+			var/client/C = g.client
+			if(C && C.admin_holder)
+				if((R_ADMIN|R_MOD) & C.admin_holder.rights) //staff don't need to see the fax twice
+					continue
+			to_chat(C, msg_ghost)
+			SEND_SOUND(C, 'sound/effects/sos-morse-code.ogg')
+
+/**
+ * Allows an admin to send a fax.
+ *
+ * Arguments:
+ * user: required, the /mob of the admin attempting to send a fax.
+ * replying_to: optional, the /mob that sent the fax the admin is replying to
+ * origin: optional, the /obj/structure/machinery/faxmachine that sent the fax the admin is replying to
+ * from_uplink: optional, the /datum/uplink that is sending the fax
+ * from_network: optional, the network from /datum/uplink that is sending the fax
+ */
+/datum/communications_network/proc/admin_fax(mob/user, mob/replying_to, obj/structure/machinery/faxmachine/origin, datum/uplink/from_uplink, from_network)
+	if(!from_uplink)
+		var/options = tgui_input_list(user, "Which uplink to appear from?", "Available Uplinks", uplinks)
```

Comment:
this is why I'd like it named something better; an admin won't know what an "uplink" is at a glance